nvflare/edge/executors/edge_dispatch_executor.py

```diff
@@ -0,0 +1,125 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import time
+from typing import Any
+
+from nvflare.apis.event_type import EventType
+from nvflare.apis.fl_context import FLContext
+from nvflare.apis.shareable import ReservedHeaderKey, ReturnCode, Shareable
+from nvflare.edge.aggregators.edge_result_accumulator import EdgeResultAccumulator
+from nvflare.edge.executors.ete import EdgeTaskExecutor
+from nvflare.edge.web.models.result_report import ResultReport
+from nvflare.edge.web.models.result_response import ResultResponse
+from nvflare.edge.web.models.task_request import TaskRequest
+from nvflare.edge.web.models.task_response import TaskResponse
+from nvflare.fuel.utils.validation_utils import check_non_negative_int, check_non_negative_number
+
+
+class EdgeDispatchExecutor(EdgeTaskExecutor):
+    """This executor dispatches tasks to edge devices and wait for the response from all devices"""
+
+    def __init__(self, wait_time=300.0, min_devices=0, aggregator_id=None):
+        EdgeTaskExecutor.__init__(self)
+
+        check_non_negative_number("wait_time", wait_time)
+        check_non_negative_int("min_devices", min_devices)
+
+        self.wait_time = wait_time
+        self.min_devices = min_devices
+        self.task_sequence = 0
+        self.task_name = None
+        self.task_id = None
+        self.task_data = None
+        self.start_time = None
+        self.devices = None
+        self.num_results = 0
+        self.aggregator_id = aggregator_id
+        self.aggregator = None
+        self.register_event_handler(EventType.START_RUN, self.setup)
+
+    def setup(self, _event_type, fl_ctx: FLContext):
+        if self.aggregator_id:
+            self.aggregator = fl_ctx.get_engine().get_component(self.aggregator_id)
+        else:
+            self.aggregator = EdgeResultAccumulator()
+
+    def convert_task(self, task_data: Shareable) -> dict:
+        """Convert task_data to a plain dict"""
+
+        return {"weights": task_data.get("weights"), "task_id": self.task_id}
```

Comment:
allow arbitrary key for general purposs (fine for basic flow, maybe add in next impl)

Reply:
This is just for the example. This class or this function should be overwritten for any other use.